### PR TITLE
Parent trace IDs

### PR DESCRIPTION
**Pull request by severinsimmler**
This PR sets the parent trace IDs from the given headers.

### CODE REVIEW

File: elastic_apm_falcon/__init__.py [hunk 1]
```diff
@@ -18,8 +20,10 @@ def __init__(self, **config):
 
     def process_request(self, req: Request, resp: Response):
         if not self.client.should_ignore_url(req.path):
+            trace_parent = TraceParent.from_headers(req.headers)
+
             # begin APM transaction
-            self.client.begin_transaction("request")
+            self.client.begin_transaction("request", trace_parent=trace_parent)
```
Comment (benedikt-budig) [inline elastic_apm_falcon/__init__.py:26]: We have to be carefult here: Can `TraceParent.from_headers` raise any exceptions and/or can it return `None`? If yes, what happens if `self.client.begin_transaction` gets `trace_parent=None`?

File: pyproject.toml [hunk 2]
```diff
@@ -12,8 +12,10 @@ elastic-apm = "^6.0.0"
 falcon = "^3.0.0"
 
 [tool.poetry.dev-dependencies]
-black = "^21.9-beta.0"
-mypy = "^0.910"
+black = "^22.6.0"
+mypy = "^0.961"
+isort = "^5.10.1"
+pycln = "^2.0.1"
```
Comment (benedikt-budig) [inline pyproject.toml:18]: shall we also add pre-commit hooks?

**Review by benedikt-budig — CHANGES_REQUESTED** [review]

File: elastic_apm_falcon/__init__.py (hunk 1 above)
Comment (severinsimmler) [reply elastic_apm_falcon/__init__.py:26]: This is copy-pasted from the [Flask agent](https://github.com/elastic/apm-agent-python/blob/be6f4edd2c0ff11f565de759fa80e5b320039683/elasticapm/contrib/flask/__init__.py#L178-L179).

Falcon's `Request.headers` should always return a dictionary (see [here](https://github.com/falconry/falcon/blob/89e19dbabb747b889caf0b4ea5256ccb77d6351f/falcon/request.py#L889-L905)). Constructing `TraceParent.from_headers(req.headers)` with an empty dictionary returns `None`. The `trace_parent` argument is by default `None` (see [here](https://github.com/elastic/apm-agent-python/blob/be6f4edd2c0ff11f565de759fa80e5b320039683/elasticapm/base.py#L295)). So nothing bad should happen here.

File: pyproject.toml (hunk 2 above)
Comment (severinsimmler) [reply pyproject.toml:18]: Yes, good idea :+1:

**Review by benedikt-budig — APPROVED** [review]
lgtm 🚀

**Pull request merged**